### ISSUE TITLE
Confusion about Training Data

### ISSUE THREAD

**Issue by zhang-qiang-github** [matterport/Mask_RCNN#110]
Does the Mask RCNN requires the per-pixel segmentation as the training data?

I have some object detection results (a rectangle for a object). Can I use those label as the training data?

Thank you for kindly answers.

Best wishes!
Qiang

**Comment by waleedka** [comment]
In the current version, yes it does require segmentation data. That's the main addition that Mask RCNN adds over other object detection algorithms. I'm working on an update to make the masks optional to allow training with bounding boxes only if desired, but it might be a few weeks before that's ready. 

In the meantime, you can either look for another model that works with bounding boxes (for example, search for Faster RCNN). Or, if you want to use this project then you can generate rectangular masks. In other words, override the load_mask() function such that it takes the bounding box coordinates and draws a mask that covers that box. This will allow you to train the model as usual and during inference you simply ignore the mask output and use the bounding box output.

**Comment by zhang-qiang-github** [comment]
Thank you for your kindly answer. 
I still have a question. If I have a object which is circle shape, I make a rectangle mask and some pixel which do not belong to the object will also be included in the rectangle mask. Like the following example.

![png](https://user-images.githubusercontent.com/29271051/33513534-c1c55b00-d77f-11e7-92e8-9594fd9f6d6a.png)

Does this situation will make the result poor?

**Comment by paulcx** [comment]
@waleedka In this case, does the mask rcnn perform better than faster rcnn in detection task for such bounding boxes? If so, we can not expect to segment the object using image-level annotation, right?

**Comment by waleedka** [comment]
@paulcx It's hard to guess, but my gut feeling is that using fake rectangular masks should not help the network be more accurate since it's not really providing any new information beyond the bounding boxes.

**Comment by WYF-Learning** [comment]
@waleedka Thany you very much for this impressive work!
When I use the pre-trained model to test my own images, I find that some background area are detected as objects. I want to use the images only have background to fine-tune the model, but I don't know how to give the  bounding boxes and masks in this situation. Could you please give me some advices? Thank you.

**Comment by teamiceberg** [comment]
Can the Mask R-CNN model be run on 3-D Matterport data? Would the presence of an additional depth dimension (using a 5-D tensor as in Matterport semantic segmentation tasks data) improve results? It looks like FAIR's Mask R-CNN model uses 2-D image sets from MS COCO for backbone network training. We are looking at a class project where we can show the impact of the depth data in adding to the performance of semantic segmentation/object detection/instance segmentation etc.? Any thoughts?
